sites/friday/src/add-services/network-storage.md

```diff
@@ -0,0 +1,142 @@
+
+It's also possible to have one app mount a `source_path` that's a subdirectory of another application's mount.
+For example:
+
+```yaml {configFile="apps"}
+{{% snippet name="app1" config="apps" %}}
+...
+mounts:
+    'web/uploads':
+        source: service
+        service: files
+        source_path: uploads
+{{% /snippet %}}
+
+{{% snippet name="app2" config="apps" globKey="false" %}}
+...
+mounts:
+    'process':
+        source: service
+        service: files
+        source_path: uploads/incoming
+    'done':
+        source: service
+        service: files
+        source_path: uploads/done
+{{% /snippet %}}
+```
+
+In this example, `app1` has access to the entire `uploads` directory by writing to `web/uploads`.
+`app2` has two mounts that it can write to: `process` and `done`.
+The `process` mount refers to the same directory as the `web/uploads/incoming` directory does on `app1`,
+and the `done` mount refers to the same directory as the `web/uploads/done` directory on `app1`.
+
+## How do I give my workers access to my main application’s files?
+
+If you need to use a worker with access to the same file mount as your web-serving app,
+define all the necessary mounts as `service` mounts.
+
+The following example assumes a Network Storage service named `files` has been defined in `{{< vendor/configfile "services" >}}`.
+Drupal files directories are shared between the `web` and `worker` instances,
+while the Drush backup directory is unique to the `web` instance.
+
+```yaml {configFile="app"}
+name: 'app'
+type: 'php:7.2'
+
+relationships:
+    database: 'db:mysql'
+
+hooks:
+  # ...
+
+web:
+    locations:
+        '/':
+            # ...
+
+disk: 1024
+
+mounts:
+    # The public and private files directories are
+    # network mounts shared by web and workers.
+    'web/sites/default/files':
+        source: service
+        service: files
+        source_path: files
+    'private':
+        source: service
+        service: files
+        source_path: private
+    # The backup, temp, and cache directories for
+    # Drupal's CLI tools don't need to be shared between web and workers.
+    # It wouldn't hurt anything to make them network
+    # shares, however.
+    '/.drush':
+        source: storage
+        source_path: drush
+    'tmp':
+        source: tmp
+        source_path: tmp
+    'drush-backups':
+        source: storage
+        source_path: drush-backups
+    '/.console':
+        source: storage
+        source_path: console
+
+# Crons run on the web container, so they have the
+# same mounts as the web container.
+crons:
+    drupal:
+        spec: '*/20 * * * *'
+        commands:
+            start: 'cd web ; drush core-cron'
+
+# The worker defined here also has the same 6 mounts;
+# 2 of them are shared with the web container,
+# the other 4 are local to the worker.
+workers:
+    queue:
+        commands:
+            start: |
+                cd web && drush queue-run myqueue
+```
```

Comment:
@chadwcarlson I'm not sure this use case is relevant in Upsun context...

Also I removed the **How can I migrate a local storage to a network storage?** section for now, because atm the only local mounts we have on Upsun are `tmp` ones, so I'm not sure if it makes sense to document how to migrate things from a `tmp` to a `service` mount? Or from a `storage` mount to a `service` one? Please let me know what you think in terms of what's relevant or not here.

Reply:
I think that this is still relevant on Upsun, but for the second case only migrating from a storage to a service mount has relevance. I don't know when someone would really need to do this, but you're right to think that tmp->service makes less sense.